src/commands/dockersupport/localDockerSupport.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { IActionContext } from "vscode-azureextensionui";
+import { localize } from '../../localize';
+import { SlotTreeItemBase } from "../../tree/SlotTreeItemBase";
+import { openUrl } from '../../utils/openUrl';
+import { requestUtils } from "../../utils/requestUtils";
+import { prompt } from "./askDockerStep";
+import { validateDockerInstalled } from './validateDockerInstalled';
+
+export const DOCKER_PROMPT_YES = "yes";
+
+/**
+ * Main Function Called to initialize the Docker flow with cloning Function App project locally from VS Code Extension
+ * @param context - behavior of action
+ * @param devContainerFolderPathUri - URI for dev container
+ * @param devContainerName - string name of dev container
+ * @param node - Function App Project
+ */
+export async function localDockerPrompt(context: IActionContext, devContainerFolderPathUri: vscode.Uri, node?: SlotTreeItemBase, devContainerName?: string): Promise<boolean> {
+    if (node && devContainerName && node.site.isLinux) {
+        if (await prompt(context) === DOCKER_PROMPT_YES) {
+            await downloadLocalDevFiles(context, devContainerFolderPathUri, devContainerName);
+            if (!(await validateDockerInstalled())) {
+                const dockerDocumentation: string = localize('documentation', 'Docker Documentation');
+                await vscode.window.showInformationMessage(localize('download', 'We noticed you don\'t have Docker installed. Check the Docker Documentation to download Docker to your system.'), dockerDocumentation,).then(async result => {
+                    if (result === dockerDocumentation) {
+                        await openUrl('https://docs.docker.com/get-docker/');
```

Comment:
no need to await on this right?

Reply:
I think so, I was thinking the user would need to click the "Docker Documentation" on the pop up or click x before continuing with the download process and opening a new window with the project and everything but I am also open to changing it to void